modules/vmseries/main.tf

```diff
@@ -86,7 +86,14 @@ resource "azurerm_virtual_machine" "this" {
   }
 
   os_profile_linux_config {
-    disable_password_authentication = false
+    disable_password_authentication = var.password == null ? true : false
+    dynamic "ssh_keys" {
+      for_each = var.ssh_key != null ? ["one"] : []
```

Comment:
We need to think about standardise how we use this statement, sometimes is:
`for_each = var.ssh_key != null ? ["one"] : []` and sometimes it is `for_each = var.ssh_key != null ? [1] : []` :)

Reply:
This is an old example, `["one"]` is used in most of the loops, hence it follows this standard. But this will be addressed in #187